wire-ios-sync-engine/Source/Use cases/OneOnOneConversationCreationStatusUseCase.swift

```diff
@@ -0,0 +1,104 @@
+//
+// Wire
+// Copyright (C) 2024 Wire Swiss GmbH
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see http://www.gnu.org/licenses/.
+//
+
+import Foundation
+import WireDataModel
+
+// sourcery: AutoMockable
+public protocol OneOnOneConversationCreationStatusUseCaseProtocol {
+
+    func invoke(userID: QualifiedID) async throws -> OneOnOneConversationCreationStatus
+
+}
+
+public enum OneOnOneConversationCreationStatus: Equatable {
+
+    case exists(protocol: MessageProtocol, established: Bool?)
+    case doesNotExist(protocol: MessageProtocol?)
```

Comment:
Do you need to return this much info? It seems like we only care if the one to one is ready and can be opened, or if it's not ready and needs to be started. If that's the case, you could simply return a Bool:

```swift
let isReady = try await checkOneOnOneIsReady.invoke(userID: userID)
```

Reply:
I needed it before to know if the conversation was meant to use `proteus` or `mls` but, since we don't care about this anymore, I'm happy to simplify. Thanks for pointing it out 🙏 

https://github.com/wireapp/wire-ios/pull/1321/commits/5bf1d89609b0ea0c3c77278eb4fd6aacc0331a89